dspace-server-webapp/src/test/java/org/dspace/app/rest/TaskRestRepositoriesIT.java

```diff
@@ -3092,4 +2943,855 @@ public void claimedTaskTest_Upload_EditMetadataOptionAllowed() throws Exception
+                                           .withAuthor("Doe, John")
+                                           .withSubject("ExtraEntry").build();
+
+        Item item1 = poolTask.getWorkflowItem().getItem();
+
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/api/workflow/pooltasks/search/findAllByItem")
+                   .param("uuid", item1.getID().toString()))
+                   .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void findAllPooltasksByItemForbiddenTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName("Parent Community")
+                                          .build();
+
+        EPerson reviewer1 = EPersonBuilder.createEPerson(context)
+                                          .withEmail("reviewer1@example.com")
+                                          .withPassword(password).build();
+
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunity)
+                                           .withName("Collection 1")
+                                           .withWorkflowGroup(1, reviewer1, admin).build();
+
+        // create a normal user to use as submitter
+        EPerson submitter = EPersonBuilder.createEPerson(context)
+                                          .withEmail("submitter@example.com")
+                                          .withPassword(password).build();
+
+        context.setCurrentUser(submitter);
+
+        PoolTask poolTask = PoolTaskBuilder.createPoolTask(context, col1, reviewer1)
+                                           .withTitle("Workflow Item 1")
+                                           .withIssueDate("2017-10-17")
+                                           .withAuthor("Smith, Donald")
+                                           .withAuthor("Doe, John")
+                                           .withSubject("ExtraEntry").build();
+
+        Item item1 = poolTask.getWorkflowItem().getItem();
+
+        context.restoreAuthSystemState();
+
+        // Only Admin has access to this end point
+
+        String tokenSubmitter = getAuthToken(submitter.getEmail(), password);
+        getClient(tokenSubmitter).perform(get("/api/workflow/pooltasks/search/findAllByItem")
+                                 .param("uuid", item1.getID().toString()))
+                                 .andExpect(status().isForbidden());
+
+        String tokenReviewer1 = getAuthToken(reviewer1.getEmail(), password);
+        getClient(tokenReviewer1).perform(get("/api/workflow/pooltasks/search/findAllByItem")
+                                 .param("uuid", item1.getID().toString()))
+                                 .andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void findAllPooltasksByItemWrongUuidTest() throws Exception {
```

Comment:
It seems there are no tests for:
* searching with a UUID not in the workflow (an archived item)
* searching with a UUID in the workflow, without pooled tasks

This seems to be missing for the other search methods as well

Reply:
the requested tests have been introduced in https://github.com/DSpace/DSpace/pull/3054/commits/3b5130c5a41d96ccf485b44a6a4d844b124261d3